From 681920a8f509915b07838f0d7b37b21ea696cd71 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <Dinh.Nguyen@freescale.com>
Date: Tue, 14 Sep 2010 20:35:52 -0500
Subject: [PATCH] ENGR00131501 fec: Fix fec_probe crash

Because of ENGR00125986, trying to access pdata structure before pdata
gets initialized.

Signed-off-by: Dinh Nguyen <Dinh.Nguyen@freescale.com>
---
 drivers/net/fec.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/fec.c b/drivers/net/fec.c
index c09cada..ff2f3af 100644
--- a/drivers/net/fec.c
+++ b/drivers/net/fec.c
@@ -1362,9 +1362,6 @@ fec_probe(struct platform_device *pdev)
 	fep = netdev_priv(ndev);
 	memset(fep, 0, sizeof(*fep));
 
-	if (!is_valid_ether_addr(fec_mac_default))
-		memcpy(fec_mac_default, pdata->mac, sizeof(fec_mac_default));
-
 	ndev->base_addr = (unsigned long)ioremap(r->start, resource_size(r));
 	fep->pdev = pdev;
 
@@ -1379,6 +1376,9 @@ fec_probe(struct platform_device *pdev)
 	if (pdata)
 		fep->phy_interface = pdata->phy;
 
+	if (pdata && !is_valid_ether_addr(fec_mac_default))
+		memcpy(fec_mac_default, pdata->mac, sizeof(fec_mac_default));
+
 	/* This device has up to three irqs on some platforms */
 	for (i = 0; i < 3; i++) {
 		irq = platform_get_irq(pdev, i);
-- 
1.5.4.4

